dashboard/app/models/script.rb

```diff
@@ -158,6 +158,20 @@ def hide_pilot_units
 
   validates :published_state, acceptance: {accept: SharedCourseConstants::PUBLISHED_STATE.to_h.values.push(nil), message: 'must be nil, in_development, pilot, beta, preview or stable'}
 
+  # Use after_save instead of validates in order to let seeding complete without checking this
+  # Since we seed scripts first, they do not yet know if they are in a unit_group and this leads
+  # to this check failing during the seed process
+  after_save :check_course_type_settings
```

Comment:
I'm not sure I follow, since it looks like we call after_save callbacks during script seed here: https://github.com/code-dot-org/code-dot-org/blob/61e0328e487582db37fb07453f629ef03496397b/dashboard/lib/services/script_seed.rb#L278

is there an earlier point where validations would be called which this tactic avoids?